frontend/appflowy_flutter/lib/workspace/presentation/widgets/view_actions_top_bar/view_actions.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+
+import 'package:appflowy/generated/flowy_svgs.g.dart';
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy/plugins/document/presentation/more/cubit/document_appearance_cubit.dart';
+import 'package:appflowy/plugins/document/presentation/more/font_size_slider.dart';
+import 'package:appflowy/startup/startup.dart';
+import 'package:appflowy/workspace/application/view/view_bloc.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
+import 'package:appflowy_popover/appflowy_popover.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra/theme_extension.dart';
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class DeleteViewAction extends StatelessWidget {
+  const DeleteViewAction({super.key, required this.view, this.mutex});
+
+  final ViewPB view;
+  final PopoverMutex? mutex;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlowyButton(
+      onTap: () {
+        getIt<ViewBloc>(param1: view).add(const ViewEvent.delete());
+        mutex?.close();
+      },
+      text: FlowyText.regular(
+        LocaleKeys.moreAction_deleteView.tr(),
+        color: AFThemeExtension.of(context).textColor,
+      ),
+      leftIcon: FlowySvg(
+        FlowySvgs.delete_s,
+        color: Theme.of(context).iconTheme.color,
+        size: const Size.square(18),
+      ),
+      leftIconSize: const Size(18, 18),
+      hoverColor: AFThemeExtension.of(context).lightGreyHover,
+    );
+  }
+}
+
+class DuplicateViewAction extends StatelessWidget {
+  const DuplicateViewAction({super.key, required this.view, this.mutex});
+
+  final ViewPB view;
+  final PopoverMutex? mutex;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlowyButton(
+      onTap: () {
+        getIt<ViewBloc>(param1: view).add(const ViewEvent.duplicate());
+        mutex?.close();
+      },
+      text: FlowyText.regular(
+        LocaleKeys.moreAction_duplicateView.tr(),
+        color: AFThemeExtension.of(context).textColor,
+      ),
+      leftIcon: FlowySvg(
+        FlowySvgs.copy_s,
+        color: Theme.of(context).iconTheme.color,
+        size: const Size.square(18),
+      ),
+      leftIconSize: const Size(18, 18),
+      hoverColor: AFThemeExtension.of(context).lightGreyHover,
+    );
```

Comment:
abstract this widget into a private widget class, then you can share the same logic among these three buttons. 

the only difference between them is the tap action and the hint text.